pkg/ingester/ingester.go

```diff
@@ -1344,6 +1344,18 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 			return true
 
 		// Map TSDB native histogram validation errors to soft errors.
+		case errors.Is(err, storage.ErrOOONativeHistogramsDisabled):
+			stats.invalidNativeHistogramCount++
+			updateFirstPartial(i.errorSamplers.nativeHistogramValidationError, func() softError {
+				return newNativeHistogramValidationError(globalerror.NativeHistogramOOODisabled, err, model.Time(timestamp), labels)
+			})
+			return true
+		case errors.Is(err, storage.ErrNativeHistogramsDisabled):
```

Comment:
I was not able to replicate this failure in a test, is this handled gracefully somewhere else?

Reply:
yes, we don't even try to ingest them if they are disabled, so no error, this is not needed: see later:
```
if nativeHistogramsIngestionEnabled {
```